File: DataAccessLibrary/Models/Bicycle.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DataAccessLibrary.Models
{
    public class Bicycle
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(100)]
        //[Column(TypeName = "varchar(100)")]
        public string Name { get; set; }

        [Required]
        public BicycleType BicycleType { get; set; }

        [Required]
        public decimal RentPrice { get; set; }

        [Required]
        public bool IsRented { get; set; }


    }
}

File: BikeRentWebApp/ClientApp/src/app/models/Bicycle.ts
import { BicycleType } from "./BicycleType";

export class Bicycle {
  id: number = 0;
  name: string = "";
  rentPrice: number = 0;
  isRented: boolean = false;
  bicycleType: BicycleType = new BicycleType();

  constructor() {

  };

}
